core/src/main/scala/cats/data/NonEmptyVector.scala

```diff
@@ -223,6 +223,8 @@ object NonEmptyVector extends NonEmptyVectorInstances {
     new NonEmptyVector(buf.result)
   }
 
+  def unapply[A](nev: NonEmptyVector[A]): Some[(A, Vector[A])] = Some((nev.head, nev.tail))
```

Comment:
@ceedubs There's some reason to prefer `Some` as the return type for extractors that can't fail—the compiler actually uses that information in some way, but I can't remember the details. I'll try to look up a reference…